src/main/java/net/minecraftforge/client/gui/config/CategoryConfigScreen.java

```diff
@@ -0,0 +1,370 @@
+            return new StringTextComponent(fallback);
+        return title;
+    }
+
+    /**
+     * This needs to make sure to split on line breaks properly otherwise the line breaks are rendered as characters.
+     */
+    static Collection<? extends ITextProperties> translateCommentWithFallback(@Nullable String translationKey, @Nullable String fallbackComment) {
+        if (translationKey != null) {
+            TranslationTextComponent comment = new TranslationTextComponent(translationKey);
+            if (!translationKey.equals(comment.getString())) {
+                comment.func_240701_a_(TOOLTIP_COMMENT_COLOR); // mergeStyle
+                return Collections.singletonList(comment);
+            }
+        }
+        if (fallbackComment == null)
+            return Collections.emptyList();
+        return Arrays.stream(fallbackComment.split("\n"))
+            .map(StringTextComponent::new)
+            .map(s -> s.func_240701_a_(TOOLTIP_COMMENT_COLOR)) // mergeStyle
+            .collect(Collectors.toList());
+    }
+
+    public static List<ITextProperties> createTooltip(ITextComponent title, @Nullable String commentTranslationKey, @Nullable String comment) {
+        List<ITextProperties> tooltip = new LinkedList<>();
+        tooltip.add(title.func_230532_e_().func_240701_a_(TOOLTIP_TITLE_COLOR));
+        tooltip.addAll(translateCommentWithFallback(commentTranslationKey, comment));
+        return tooltip;
+    }
+
+    @Override
+    protected ConfigElementList makeConfigElementList() {
+        return new CategoryConfigElementList(this, field_230706_i_, categoryInfo);
+    }
+
+    // TODO: Make POJO?
+    public interface ConfigCategoryInfo {
+
+        static ConfigCategoryInfo of(Supplier<Collection<String>> elements, Function<String, Object> getValue, Function<String, Object> getSpec, Function<String, String> getCategoryComment) {
+            return new ConfigCategoryInfo() {
+
+                @Override
+                public Collection<String> elements() {
+                    return elements.get();
+                }
+
+                @Override
+                public Object getValue(String key) {
+                    return getValue.apply(key);
+                }
+
+                @Override
+                public Object getSpec(String key) {
+                    return getSpec.apply(key);
+                }
+
+                @Override
+                public String getCategoryComment(String key) {
+                    return getCategoryComment.apply(key);
+                }
+            };
+        }
+
+        Collection<String> elements();
+
+        /**
+         * Warning: Can return null if you call it with a key that isn't in the collection returned from {@link #elements()}
+         *
+         * @return Either a {@link Config} (if the key corresponds to a category) or a {@link ConfigValue} (if the key corresponds to a value).
+         */
+        Object getValue(String key);
```

Comment:
Can we use DFU's Either here to make things a little cleaner?

Reply:
I can’t find out what a DFU is, I assume it’s not a Device Firmware Update which is all google will show me.

Reply:
Data Fixer-Upper, it's a Mojang tool that... is hard to explain.

Reply:
There is a class in the DFU lib that's called Either, it can be used instead of raw object if exactly two completly different types are possible